### PR TITLE
KOTORBASE: Extract PC movement logic to a separate class

### PR DESCRIPTION
**Pull request by seedhartha**

### CODE REVIEW

File: src/engines/kotorbase/module.h [hunk 1]
```diff
@@ -291,12 +292,10 @@ class Module : public Object, public ObjectContainer {
 	bool _freeCamEnabled;
 	uint32 _prevTimestamp;
 	float _frameTime;
-	float _forwardBtnPressed;
-	float _backwardsBtnPressed;
-	bool _pcRunning;
 	bool _pcPositionLoaded;
 	bool _inDialog;
 	float _cameraHeight;
+	Common::ScopedPtr<PlayerController> _playerController;
```
Comment (DrMcCoy) [inline src/engines/kotorbase/module.h]: Does it have to be a pointer at all?

Comment (seedhartha) [reply src/engines/kotorbase/module.h]: No, I've removed the pointer.

**Comment by DrMcCoy** [comment]
You can't move Trask anymore now.

In the current master, when trying to move after adding Trask, I get this sanitizer error: https://gist.github.com/DrMcCoy/cb7fd05a84272b45738fd966e78ae65c

In the current master, when equipping a short sword, I get this sanitizer error: https://gist.github.com/DrMcCoy/b42fe70b3d45aae11b414e55590022b6

**Comment by seedhartha** [comment]
Fixed Trask not being able to move. The other issues I will try to fix as part of the other PR.

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Pull request merged**